tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/debug/NodeJSAppDebugTest.java

```diff
@@ -0,0 +1,338 @@
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * 
+ * @author psrna
+ *
+ */
+@OpenPerspective(DebugPerspective.class)
+@RunWith(RedDeerSuite.class)
+@OCBinary
+@RequiredBasicConnection
+@RequiredProject
+@RequiredService(service = OpenShiftResources.NODEJS_SERVICE, template = OpenShiftResources.NODEJS_TEMPLATE)
+public class NodeJSAppDebugTest {
+
+	@InjectRequirement
+	private OpenShiftProjectRequirement requiredProject;
+	@InjectRequirement
+	private OpenShiftConnectionRequirement requiredConnection;
+	@InjectRequirement
+	private OpenShiftServiceRequirement requiredService;
+	private OpenShift3Connection connection;
+	private OpenShiftProject project;
+	private ServerAdapter adapter;
+
+	private static int BREAKPOINT_LINE = 85;
+	private static String APP_URL = "http://nodejs-example-test-project.rhel-cdk.10.1.2.2.xip.io/pagecount";
+	private static String VAR_db = "db";
+	private static String VAR_VALUE_db = "null";
+	private static String VAR_initDb = "initDb";
+	private static String VAR_VALUE_initDb = "[Function]";
+	private static String VAR_require = "require";
+	private static String VAR_VALUE_require = "[Function]";
+
+	@Before
+	public void setUp() {
+		TestUtils.setUpOcBinary();
+		// TODO use OpenShiftResources.NODEJS_GIT_NAME (once merged)
+		TestUtils.cleanupGitFolder("nodejs-ex");
+		OpenShiftExplorerView explorer = new OpenShiftExplorerView();
+		explorer.open();
+		this.connection = explorer.getOpenShift3Connection(requiredConnection.getConnection());
+		this.project = connection.getProject(requiredProject.getProjectName());
+		this.project.expand();
+
+		// import application
+		this.project.getService(OpenShiftResources.NODEJS_SERVICE).select();
+		new ContextMenu(OpenShiftLabel.ContextMenu.IMPORT_APPLICATION).select();
+		new DefaultShell(OpenShiftLabel.Shell.IMPORT_APPLICATION);
+		new FinishButton().click();
+		new WaitWhile(new ShellWithTextIsAvailable(OpenShiftLabel.Shell.IMPORT_APPLICATION));
+
+		// setup server adapter
+		explorer.activate();
+		this.project.getService(OpenShiftResources.NODEJS_SERVICE).select();
+		new ContextMenu(OpenShiftLabel.ContextMenu.NEW_ADAPTER_FROM_EXPLORER).select();
+		new DefaultShell(OpenShiftLabel.Shell.SERVER_ADAPTER_SETTINGS);
+		new FinishButton().click();
+
+		new WaitWhile(new ShellWithTextIsAvailable(""), TimePeriod.LONG);
+		new WaitWhile(new JobIsRunning(), TimePeriod.VERY_LONG);
+
+		assertTrue("OpenShift 3 server adapter was not created.",
+				new ServerAdapterExists(Version.OPENSHIFT3, OpenShiftResources.NODEJS_SERVICE).test());
+
+		// restart in debug
+		this.adapter = new ServerAdapter(Version.OPENSHIFT3, OpenShiftResources.NODEJS_SERVICE);
+		this.adapter.select();
+		new ContextMenu("Restart in Debug").select();
+		new WaitWhile(new JobIsRunning(), TimePeriod.VERY_LONG);
+
+	}
+
+	@After
+	public void clean() {
+		ShellHandler.getInstance().closeAllNonWorbenchShells();
```

Comment:
The test will probably fail anyway. But I wanted to make sure its clean for the following tests. Maybe I should put it into @Before?